File: 48_count_of_digit/48_count_of_digit.dart
import 'dart:io';

// method to calculate the digit count of number
int digitCount(int num){
  int cnt = 0;
  if(0 == num){
    return 1;
  }
  while (num > 0){
    cnt += 1;
    num ~/= 10;
  }
  return cnt;
}

// drivr program
void main() {
  try {
    while (true) {
      stdout.write("Enter number in integer form : ");
      int num = int.parse(stdin.readLineSync()!);

      if (num < 0){
        num *= -1;
      }

      int result = digitCount(num);
      print("Count of digint in $num ===> $result");

      stdout.write("Do you want to continue the program (y/n) : ");
      String choice = stdin.readLineSync()!.toLowerCase();
      if (choice == 'n') {
        print("Program finished!");
        break;
      }
    }
  } on FormatException catch (e) {
    print("Invalid input : $e");
  } on UnsupportedError catch (e) {
    print("Divide by zero exception : $e");
  } on Exception catch (e) {
    print("Exception caught : $e");
  }
}

File: 48_count_of_digit/48_count_of_digit.py
# Python progrsm to count the numebr the digit in a particular number

# method 1 - using modulus operator and brut force
try:
    n=int(input("Enter the number :"))
    cnt=0
    while n>0:
        n=n//10
        cnt=cnt+1
    print(cnt)
except:
    print("Enter a valid input.")

# method 2 - using len() method
try:
    n=int(input("Enter the number :"))
    print(len(str(n)))
except:
    print("Enter a valid input.")

# method 3 - using recursion
try:
    n=int(input("Enter the number :"))
    cnt=1
    def digit_count(a):
        if a==0:
            return print(end="")
        global cnt
        cnt+=1
        digit_count(a//10)
    digit_count(n//10)
    print(cnt)
except:
    print("Enter a valid input.")

# method 4 - using module based approach

def digit_count(num):
    if (0 == num):
        return 1
    cnt = 0
    while num > 0:
        cnt+=1
        num//=10
    return cnt

try:
    while (True):
        
        num = int(input("Enter number : "))

        if(num < 0):
            num *= -1

        result = digit_count(num)
        print(f"No of digit in {num} ===> {result}")

        choice = input("Do you want to continue the program (y/n) : ")
        if(choice.lower() == "n"):
            print("Program finished!")
            break

except ValueError as e:
    print(f"Invalid Input : {e}")

except ZeroDivisionError as e:
    print(f"Zero division exception : {e}")

except Exception as e:
    print(f"Exception caught : {e}")
